File: stack/infixToPostfix.rs
use std::collections::HashMap;

fn main()
{

let precedence : HashMap <char , u8> = HashMap::from([('+',0),('-',0),('*',1),('/',1)]);

let mut input_line = String::new();
println!("Enter the infix expression :");
std::io::stdin().read_line(&mut input_line).expect("Error while reading");

let mut operator_stack : Vec<char> = Vec::new();

let mut output_stream = String::from ("Corresponding Postfix : ");
let mut val : u8;
let mut operator_buffer : char;

for char_val in input_line.chars()
{
 val = char_val as u8;
 if val == 13 || val == 10 {continue;}
 
 match val 
 {
  48..=57 => {val -=48 ; output_stream.push_str(&val.to_string());},
  42 | 43 | 45 | 47 => {
   operator_buffer = val as char;
   loop {
     if operator_stack.len() == 0 {operator_stack.push(operator_buffer); break;}
     else if precedence.get(&operator_stack[operator_stack.len()-1]) >= precedence.get(&operator_buffer)
     {output_stream.push_str(&operator_stack[operator_stack.len()-1].to_string()); 
      operator_stack.pop();
     }
     else {operator_stack.push(operator_buffer); break;}
   } 
  }
   
  _ => {println!("Error in the infix notation supplied "); std::process::exit(1);}
 }
}

while operator_stack.len() != 0
{
 output_stream.push_str(&operator_stack[operator_stack.len()-1].to_string());
 operator_stack.pop();
}



println!("{}" , output_stream);
}

File: stack/infixToPostfix.cpp
#include <iostream>
#include <string>
#include <stack>
#include <unordered_map>

int main()
{
std::cout << "Enter the infix expression : " << std::endl;
std::string data;
std::stack <char> operator_stack;
std::unordered_map <char , int> operatorPrecedence = {
    {'+',0},
    {'-',0},
    {'*',1},
    {'/',1},
    {'%',1}, 
};


std::cin >> data;
std::string result = "";
char operator_buffer;
for (int val : data)
{
    if (val >= 48 && val <= 57) result+=val;
    // 42 | 43 | 45 | 47
    if (val == 37 || val == 42 || val == 43 || val == 45 || val == 47)
    {
        operator_buffer = (char) val;
        while (true){
         if (operator_stack.size() == 0) {operator_stack.push(operator_buffer); break;}
     else if (operatorPrecedence[operator_stack.top()] >= operatorPrecedence[operator_buffer])
     {result+=operator_stack.top(); 
      operator_stack.pop();
     }
     else {operator_stack.push(operator_buffer); break;}
        }
    
    }
}


while (operator_stack.size() != 0)
{
 result+=operator_stack.top();
 operator_stack.pop();
}

std::cout << "this is the result : " << result;


return 0;
}